### PR TITLE
Add metrics to discv5

### PR DESCRIPTION
**Pull request by dshulyak**

### CODE REVIEW

**Comment by mandrigin** [comment]
Looks good, but we need a patch for it, right?

File: vendor/github.com/ethereum/go-ethereum/p2p/discv5/net.go [hunk 1]
```diff
@@ -40,8 +40,9 @@ var (
 )
 
 const (
-	autoRefreshInterval   = 1 * time.Hour
-	bucketRefreshInterval = 1 * time.Minute
+	autoRefreshInterval = 1 * time.Hour
+	// spams the network, we do enough lookups when topic is being search
```
Comment (divan) [inline vendor/github.com/ethereum/go-ethereum/p2p/discv5/net.go]: typo: `s/being search/being searched/`

File: vendor/github.com/ethereum/go-ethereum/p2p/discv5/net.go [hunk 2]
```diff
@@ -40,8 +40,9 @@ var (
...
+	bucketRefreshInterval = 120 * time.Minute
```
Comment (divan) [inline vendor/github.com/ethereum/go-ethereum/p2p/discv5/net.go]: Can you elaborate on where this estimation comes from? Better to put it into PR description, so it's discoverable later on.

**Review by divan — APPROVED** [review]
LGTM, plus couple of nitpicks:
 - tuning and metrics seems to very unrelated things, so it's better to separate them into different PRs (easier to review, understand, revert if needed, etc)
 - geth patch needed

File: vendor/github.com/ethereum/go-ethereum/p2p/discv5/metrics.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package discv5
+
+import "github.com/ethereum/go-ethereum/metrics"
+
+var (
+	ingressTrafficMeter = metrics.NewMeter("discv5/InboundTraffic")
```
Comment (themue) [inline vendor/github.com/ethereum/go-ethereum/p2p/discv5/metrics.go:6]: Why `egress` = `Inbound` and `egress` = `Outbound`? Would prefer consistent naming internal and external.

File: vendor/github.com/ethereum/go-ethereum/p2p/discv5/net.go (hunk 2 above)
Comment (themue) [reply vendor/github.com/ethereum/go-ethereum/p2p/discv5/net.go]: Best in PR description plus a short comment in code for later maintainers don't looking into the PR history.

**Review by themue — CHANGES_REQUESTED** [review]

File: vendor/github.com/ethereum/go-ethereum/p2p/discv5/metrics.go (hunk 3 above)
Comment (dshulyak) [reply vendor/github.com/ethereum/go-ethereum/p2p/discv5/metrics.go:6]: this is consistent with p2p metrics https://github.com/ethereum/go-ethereum/blob/master/p2p/metrics.go#L28-L31

Comment (themue) [reply vendor/github.com/ethereum/go-ethereum/p2p/discv5/metrics.go:6]: IC, thanks. Still feels strange. 😁

**Comment by dshulyak** [comment]
Removed tuning cause stopping discv5 enough to keep traffic under control

**Review by adambabik — APPROVED** [review]

**Comment by adambabik** [comment]
@themue is it ok to merge now?

**Review by themue — APPROVED** [review]

**Pull request merged**